### DISCUSSION TITLE
Include open-vm-tools package in Home Assistant OS's OVA image for VMware virtual machine hypervisors?

### DISCUSSION THREAD

**Discussion by Hedda** [home-assistant/operating-system#2693] [Ideas]
Please consider including Open VM Tools (open-vm-tools) in Home Assistant OS's OVA-image for VMware's  virtual machine hypervisor:

https://github.com/vmware/open-vm-tools

https://raw.githubusercontent.com/vmware/open-vm-tools/master/README.md

(Great if it could be included in alternative images for "VMware ESXi/vSphere" -> https://www.home-assistant.io/installation/alternative ).

open-vm-tools is VMware's open-source (GPL v2 licensed) variant of "VMware Tools" meant to be installed on Linux/BSD/Solaris-based operating-systems running on virtual machines in VMware vSphere ESXi, VMware Workstation, VMware Fusion virtualization hypervisors. Its purpose is the simplify lifecycle management and support for guest operating systems.

Personally I like to see Open VM Tools included primarily because it allows you to shrink the disk but it also has many other features.

"**_open-vm-tools is a set of services and modules that enable several features in VMware products for better management of, and seamless user interactions with, guests. It includes kernel modules for enhancing the performance of virtual machines running Linux or other VMware supported Unix like guest operating systems._** "

**_open-vm-tools enables the following features in VMware products:_**

- **_The ability to perform virtual machine power operations gracefully._**
- **_Execution of VMware provided or user configured scripts in guests during various power operations._**
- **_The ability to run programs, commands and file system operation in guests to enhance guest automation._**
- **_Authentication for guest operations._**
- **_Periodic collection of network, disk, and memory usage information from the guest._**
- **_Generation of heartbeat from guests to hosts so VMware's HA solution can determine guests' availability._**
- **_Clock synchronization between guests and hosts or client desktops._**
- **_Quiescing guest file systems to allow hosts to capture file-system-consistent guest snapshots._**
- **_Execution of pre-freeze and post-thaw scripts while quiescing guest file systems._**
- **_The ability to customize guest operating systems immediately after powering on virtual machines._**
- **_Enabling shared folders between host and guest file systems on VMware Workstation and VMware Fusion._**
- **_Copying and pasting text, graphics, and files between guests and hosts or client desktops._**

https://kb.vmware.com/s/article/2073803

https://docs.vmware.com/en/VMware-Tools/index.html

**Comment by agners** [comment]
OpenVMTools is pre-installed in the OVA image, see https://github.com/home-assistant/operating-system/blob/dev/buildroot-external/configs/ova_defconfig#L111.

**Comment by deadrubberboy** [comment]
I just moved from UTM to VMWare Fusion on Mac (Apple silicon).  I’m getting an error that VM ware Tools is not installed when I go to reboot.  How can I add?

**Reply by volkerrichert** [reply]
VMware Fusion uses requires the vmdk image, not the ova image

**Reply by mikeschmick** [reply]
I sorted it out. Just selected to shut down the VM on reboot.
